### ISSUE TITLE
Are pretrained models intended to be downlaoded into /home by torch.hub?

### ISSUE THREAD

**Issue by johnohagan** [ultralytics/yolov5#3889]
Hi

I'm a python user using pretrained object detection for an art project, so not a developer! 

I was using cv2.dnn with yolov3 but recently switched to torch.hub with yolov5. I love how easy it is to use, eg model = torch.hub.load(...), model.conf = 0.5, results = model(image).xyxyn, etc, exactly how software should be! 
 
Something I've noticed when calling torch.hub.load from the ultralytics repo for the first time is that the model zipfiles are downloaded into my home directory. Is this intentional? This is not a big deal obviously, but other torch.hub repos seem to put these in a directory in ~.cache/torch/hub along with the other required files, which seems neater to me.

I added this line to my hubconf.py that fixes this, but obviously it disappears if I call anything with force_reload:

name = ''.join((str(Path(__file__).parent), '/', name))

Should I submit this as a bug report / feature request?

Thanks

**Comment by glenn-jocher** [comment]
@johnohagan hmm that's an interesting idea. I'm not sure if people actually want to see these model.pt files or if they want them hidden away.

But sure please submit a PR, thanks!

See our [Contributing Guide](https://docs.ultralytics.com/help/contributing/) if you have questions on how to submit a PR.

**Comment by glenn-jocher** [comment]
@johnohagan I think the path code you want is something like this, but you should also be aware that passing a path will download the model to that path also, which is a feature we need to retain:
```python
model = torch.hub.load('ultralytics/yolov5', 'custom', 'path/to/download/model.pt')
```
so you probably want to check that name does not include any directories before assigning it to the cache directory with something like:
```python
name = Path(__file__).parent / name
```

EDIT: Above comments mainly apply to `torch.hub.load('ultralytics/yolov5', 'custom', ...)`

**Comment by glenn-jocher** [comment]
@johnohagan FYI hubconf.py updated slightly in https://github.com/ultralytics/yolov5/pull/3894 with unrelated changes today.

**Comment by johnohagan** [comment]
Thanks, I submitted that pull request. I should have made it clear that the line I added was to the _create function and modifies the "name" argument passed to that function.

Since then I had a quick look at the where the pytorch/vision repo puts .pt(h) files, and it's ~/.cache/torch/hub/checkpoints. If that was considered better the line would be 
`
name = ''.join((str(Path(__file__).parent.parent), '/checkpoints/', name))`

I've also just learbed from your comment that I don't need the ''.join for Path objects so I'll fix that either way.
